api/storage/file.go

```diff
@@ -102,6 +106,14 @@ func (p PresignUrlOptions) isValid() error {
 	return nil
 }
 
+func (o *fileImpl) UploadUrl(expiry int) (string, error) {
+	return o.SignUrl(PresignUrlOptions{Expiry: expiry, Mode: ModeWrite})
+}
+
+func (o *fileImpl) DownloadUrl(expiry int) (string, error) {
+	return o.SignUrl(PresignUrlOptions{Expiry: expiry, Mode: ModeRead})
+}
+
 func (o *fileImpl) SignUrl(opts PresignUrlOptions) (string, error) {
 	if err := opts.isValid(); err != nil {
 		return "", errors.NewWithCause(codes.InvalidArgument, "invalid options", err)
```

Comment:
in nodejs this method has been marked as deprecated in favour of the Upload/Download url shortcut methods (and will be made private in the future). Wanted to get your thoughts on doing the same for go to reduce the amount of documentation required for us to maintain.

Reply:
well given that is not officially released, I can just make it private.